src/ua.c

```diff
@@ -1982,6 +2021,59 @@ struct sipevent_sock *uag_sipevent_sock(void)
 }
 
 
+static bool uri_match_transport(const struct uri *accu, enum sip_transp tp)
+{
+	struct pl pl;
+	int err;
+
+	err = msg_param_decode(&accu->params, "transport", &pl);
+	if (err)
+		return true;
+
+	return !pl_strcasecmp(&pl, sip_transp_name(tp));
+}
```

Comment:
I think here it is better to compare the sip transport tp instead of comparing the string.

1. decode transport parameter
2. convert from string to `enum sip_transp`
3. compare it with tp